lib/test_formulae.rb

```diff
@@ -132,38 +134,43 @@ def download_artifact_from_previous_run!(artifact_name, dry_run:)
         check_suite_nodes = response.dig("repository", "object", "checkSuites", "nodes")
         return if check_suite_nodes.blank?
 
-        wanted_artifact = artifact_metadata(check_suite_nodes, github_repository, "pull_request",
-                                            "CI", "conclusion", artifact_name)
-        # If we didn't find the artifact that we wanted, fall back to the `event_payload` artifact.
-        wanted_artifact ||= artifact_metadata(check_suite_nodes, github_repository, "pull_request_target",
-                                              "Triage tasks", "upload-metadata", "event_payload")
-        return if wanted_artifact.blank?
+        wanted_artifacts = artifact_metadata(check_suite_nodes, github_repository, "pull_request",
+                                             "CI", "conclusion", artifact_pattern)
+        wanted_artifacts_pattern = artifact_pattern
+        if wanted_artifacts.empty?
+          # If we didn't find the artifacts that we wanted, fall back to the `event_payload` artifact.
+          wanted_artifacts = artifact_metadata(check_suite_nodes, github_repository, "pull_request_target",
+                                               "Triage tasks", "upload-metadata", "event_payload")
+          wanted_artifacts_pattern = "event_payload"
+        end
+        return if wanted_artifacts.empty?
 
-        wanted_artifact_name = wanted_artifact.fetch("name")
-        if @downloaded_artifacts[sha].include?(wanted_artifact_name)
-          opoo "Already tried #{wanted_artifact_name} from #{sha}, giving up"
+        if @downloaded_artifacts[sha].include?(wanted_artifacts_pattern)
+          opoo "Already tried #{wanted_artifacts_pattern} from #{sha}, giving up"
           return
         end
 
-        @downloaded_artifacts[sha] << wanted_artifact_name
-        cached_event_json&.unlink if wanted_artifact_name == "event_payload"
+        @downloaded_artifacts[sha] << wanted_artifacts_pattern
+        cached_event_json&.unlink if wanted_artifacts_pattern == "event_payload"
 
-        ohai "Downloading #{wanted_artifact_name} from #{sha}"
-        download_url = wanted_artifact.fetch("archive_download_url")
-        artifact_id = wanted_artifact.fetch("id")
+        ohai "Downloading artifacts matching pattern #{wanted_artifacts_pattern} from #{sha}"
+        wanted_artifacts.each do |artifact|
+          download_url = artifact.fetch("archive_download_url")
+          artifact_id = artifact.fetch("id")
 
-        require "utils/github/artifacts"
+          require "utils/github/artifacts"
 
-        artifact_cache.mkpath
-        artifact_cache.cd do
-          GitHub.download_artifact(download_url, artifact_id.to_s)
+          artifact_cache.mkpath
+          artifact_cache.cd do
```

Comment:
You could probably hoist most of this logic outside the `each` loop (i.e. `require`, `#mkpath`, and `#cd`)